src/goto-programs/loop_unroll.cpp

```diff
@@ -0,0 +1,213 @@
+#include <goto-programs/loop_unroll.h>
+
+bool unsound_loop_unroller::runOnLoop(loopst &loop, goto_programt &goto_program)
+{
+  int bound = get_loop_bounds(loop);
+  if(bound <= 0)
+    return false; // Can't unroll this
+
+  // Get loop exit goto number
+  goto_programt::targett loop_exit = loop.get_original_loop_exit();
+  // TODO: If we want a sound version: loop_exit++ so we have the skip instruction
+  if(loop_exit != goto_program.instructions.begin())
+  {
+    goto_programt::targett t_before = loop_exit;
+
+    if(t_before->is_goto() && is_true(t_before->guard))
```

Comment:
Add a comment here to explain this condition.

Reply:
I removed the condition.